src/pages/chat/Chat.tsx

```diff
@@ -141,6 +153,88 @@ export default function Chat() {
   const [previousPath, setPreviousPath] = useState<string>("");
   const navigate = useKeepQueryParamsNavigate();
   const { isXXS, isXS, isS } = useBreakpoints();
+  const { showModal, modalTypes } = useModal();
+  const { bosonXmtp } = useChatContext();
+  const notifiConfig = getNotifiConfig();
+
+  // TODO: ideally if notifiConfig is null, do not call useNotifiClient()
+
+  const notifiClient = useNotifiClient({
+    dappAddress: notifiConfig?.dappId || "",
+    env: "Development",
+    walletPublicKey: address as string,
+    walletBlockchain: notifiConfig?.chain || "ETHEREUM"
+  });
+  const fetchNotifiData = (data: NotifiClientData | null) => {
+    const xmtpAlert = data?.alerts?.find(
+      (alert: { filter?: { filterType?: string } }) =>
+        alert.filter?.filterType === "WEB3_CHAT_MESSAGES"
+    );
+    const xmtpSources = xmtpAlert?.sourceGroup?.sources?.filter(
+      (source: unknown) => (source as { type?: string }).type === "XMTP"
+    );
+    const sourceTopics = xmtpSources?.map(
+      (source: unknown) => (source as { name?: string }).name
+    ) as string[];
+    setAlreadyRegisteredTopics(sourceTopics);
```

Comment:
```suggestion
    const sourceTopics =
      xmtpSources?.map((source) => source?.name).filter(isTruthy) ?? [];
      // TODO: verify source topics should not be empty if alreadyRegisteredTopics was already empty?
    setAlreadyRegisteredTopics(sourceTopics);
```

Reply:
This is the change that leads to the rerender loop.
To avoid that, I've define a constant `const EMPTY_STRING_ARRAY: string[] = [];` and I'm using that fallback value instead of `[]` (that is, I guess, each time a different instance of an empty Array, then it's triggering the hook indefinitely...)